packages/core/core/src/utils.js

```diff
@@ -51,3 +52,35 @@ export function registerCoreWithSerializer() {
 
   coreRegistered = true;
 }
+
+export function getPublicId(
+  id: string,
+  alreadyExists: string => boolean,
```

Comment:
Wouldn't this introduce the possibility of generating different id's for each build?
Weren't ids supposed to remain the same across builds for caching?